src/BitStringAddresses/bosefs.jl

```diff
@@ -331,11 +343,29 @@ end
 function hopnextneighbour(b::SingleComponentFockAddress, i)
     src = find_occupied_mode(b, (i + 1) >>> 0x1)
     dst = find_mode(b, mod1(src.mode + ifelse(isodd(i), 1, -1), num_modes(b)))
-
     new_b, val = excitation(b, (dst,), (src,))
     return new_b, val
 end
 
+function hopnextneighbour(b::SingleComponentFockAddress, i, ::Val{boundary_condition}) where {boundary_condition}
+    src = find_occupied_mode(b, (i + 1) >>> 0x1)
+    dst = find_mode(b, mod1(src.mode + ifelse(isodd(i), 1, -1), num_modes(b)))
+    new_b, val = excitation(b, (dst,), (src,))
+    if (find_occupied_mode(b,1).mode==1 && i==2) || 
+        (find_occupied_mode(b,num_occupied_modes(b)).mode == num_modes(b) 
+        && i == (2*num_occupied_modes(b)-1))
+        if boundary_condition == :twisted
+            return new_b,-val
+        elseif boundary_condition == :hard_wall
+            return new_b, 0.0
+        else
+            return new_b, val
+        end
+    else
+        return new_b, val
+    end
+end
```

Comment:
This is good but it will (sometimes) be much faster like this. Hard to explain why.
```suggestion
function hopnextneighbour(
    b::SingleComponentFockAddress, i, ::Val{boundary_condition}
) where {boundary_condition}
    src = find_occupied_mode(b, (i + 1) >>> 0x1)
    dir = ifelse(isodd(i), 1, -1)
    dst = find_mode(b, mod1(src.mode + dir, num_modes(b)))
    new_b, val = excitation(b, (dst,), (src,))

    on_boundary = src.mode == 1 && dir == -1 || src.mode == num_modes(b) && dir == 1

    if boundary_condition == :twisted && on_boundary
        return new_b, -val
    elseif boundary_condition == :hard_wall && on_boundary
        return new_b, 0.0
    else
        return new_b, val
    end
end
```